zerver/actions/streams.py

```diff
@@ -447,6 +462,14 @@ def send_stream_creation_events_for_previously_inaccessible_streams(
     stream_ids = set(altered_user_dict.keys())
     recent_traffic = get_streams_traffic(stream_ids, realm)
 
+    setting_group_ids = set()
+    for stream_id in stream_ids:
+        stream = stream_dict[stream_id]
+        for setting_name in Stream.stream_permission_group_settings:
+            setting_group_ids.add(getattr(stream, setting_name + "_id"))
+
+    setting_groups_dict = get_setting_values_for_group_settings(list(setting_group_ids))
```

Comment:
There can be a slight modification where we call this code only if there are actual users to nofify and not make this query unnecessarily. We would also need to make sure we only compute this once.

Reply:
Yeah that seems like a good optimization. Can probably be a follow-up, but we should be sure to not lose track of it.